src/Handlers.cpp.Rt

```diff
@@ -897,49 +897,108 @@ class cbFailcheck: public Callback {
 		reg.dx = solver->region.dx;
 		reg.dy = solver->region.dy;
 		reg.dz = solver->region.dz;
-               	pugi::xml_attribute attr = node.attribute("dx");
-                if (attr) {
-                       	reg.dx = solver->units.alt(attr.value());
-                }
-                attr = node.attribute("dy");
-                if (attr) {
-                        reg.dy = solver->units.alt(attr.value());
-                }
-                attr = node.attribute("dz");
-                if (attr) {
-                       	reg.dz = solver->units.alt(attr.value());
-                }
+	
+
+
+        pugi::xml_attribute attr = node.attribute("dx");
+        if (attr) {
+            reg.dx = solver->units.alt(attr.value());
+        }
+        attr = node.attribute("dy");
+        if (attr) {
+            reg.dy = solver->units.alt(attr.value());
+        }
+        attr = node.attribute("dz");
+        if (attr) {
+            reg.dz = solver->units.alt(attr.value());
+        }
+
+
+        attr = node.attribute("nx");
+        if (attr) {
+            reg.nx = solver->units.alt(attr.value());
+        }
+        attr = node.attribute("ny");
+        if (attr) {
+            reg.ny = solver->units.alt(attr.value());
+        }
+        attr = node.attribute("nz");
+        if (attr) {
+            reg.nz = solver->units.alt(attr.value());
+        }
+
 		rkept = 1;
 		return 0;
 	}
 	int DoIt() {
 		Callback::DoIt();
-        	int ret = 0;
-		int cond,fin;
+       	int ret = 0;
+		int cond,fin,go;
 		fin = false;
 		cond = false;
+        go = false;
+
+        pugi::xml_attribute comp = node.attribute("what");
+        std::vector<std::string> components;
+        std::vector<real_t*> buffers;
```

Comment:
I think, default components should have one element with value "ALL". So, that basic  Failcheck Iterations="1000/> will be just checking all quantities. 

But,  as a matter of fact, we already have class for operating with quantities/names in Handlers, namely `name_set` in utills. 
So it is much easier to do: 
```c
nameset s;
pugi::xml_attribute comp = node.attribute("what");
if (comp) {
        s.add_from_string(attr.value(),',');
  }
 else {
     s.add_from_string("all",',');
 }
```
and in then in the loop, do stuff only for quantities specified,as : 
```c  
<?R for (q in rows(Quantities)) { ifdef(q$adjoint); ?> 
if (s->in("<?%s q$name ?>")) 
{
// do stuff 
}
<?R }; ifdef(); ?>
```
This looks much nicer, eliminate raw string manipulation and also "go" variable will no longer be needed.